proto/src/determined/checkpoint/v1/checkpoint.proto

```diff
@@ -29,6 +31,34 @@ enum State {
   STATE_DELETED = 4;
 }
 
+
+// TODO extended checkpoint instead?
+
+// CheckpointBase is an artifact created by a trial during training.
+message CheckpointBase {
```

Comment:
So, sort of like validations, but explicitly this time, this is just the slice of a checkpoint that you need for the trials endpoint. I'll think on the best solution for both of these over the weekend.

gRPC certainly makes you think about having well-defined models more than just jsonify'ing a sql query, ha.

Reply:
>  this is just the slice of a checkpoint that you need for the trials endpoint.

maybe the other Checkpoint definition has some additional data bundled in and we want a smaller set :thinking: let's chat. this is the last remaining part here.